packages/ui-causal-graph-editor/src/shared/causal-graph-store.tsx

```diff
@@ -138,8 +139,10 @@ export const GraphReducer: Reducer<GraphState, GraphAction> = (state, action) =>
         }
 
         case GraphActionType.ADD_EDGE: {
+            const editorMode = draft.editorMode ?? (isDag(draft.graph) ? EditorMode.DEFAULT : EditorMode.PAG_VIEWER);
```

Comment:
Isn't editor mode initialized in `useCausalGraphEditor`? Should it be defaulted there? Or should it be set as part of the `INIT` action? This works but it really shouldn't be undefined here

Reply:
Thank you, I had initially tried to have it on useCausalGraphEditor but because the graph needed to be parsed first I got into a chicken egg situation, putting it on the initialiser was a good idea, hadn't occurred to me that I could do that :sweat_smile: